src/Utilities/SetHistory.js

```diff
@@ -2,10 +2,12 @@ import queryString from 'query-string';
 import { ASC, DESC } from '../constants';
 
 export function setHistory(
-    history, systemIds = [], baselineIds = [], hspIds = [], referenceId, activeFactFilters = [], factFilter, stateFilters, factSort, stateSort
+    history, systemIds = [], baselineIds = [], hspIds = [], referenceId, activeFactFilters = [], factFilter, factTypeFilters, stateFilters, factSort,
+    stateSort
 ) {
     let nameFilters = [ ...activeFactFilters, ...factFilter && !activeFactFilters.includes(factFilter) ? [ factFilter ] : [] ];
     let filterState = [ ...stateFilters?.filter(({ selected }) => selected)?.map(({ filter }) => filter?.toLowerCase()) || [] ];
+    let filterFactType = [ ...factTypeFilters?.filter(({ selected }) => selected)?.map(({ filter }) => filter?.toLowerCase()) || [] ];
```

Comment:
I believe the same result can be achieved without array destruction. I have checked as following:

let factTypeFilters = [{ id: 1, selected:true, filter: 'a'}, {id:2, selected:false,  filter: 'b'}];
const result = factTypeFilters?.filter(({ selected }) => selected)?.map(({ filter }) => filter?.toLowerCase()) || [];
console.log(result);